flink-runtime/src/test/java/org/apache/flink/runtime/taskmanager/TaskManagerLocationTest.java

```diff
@@ -198,4 +199,24 @@ public void testGetHostname2() {
 			fail(e.getMessage());
 		}
 	}
+
+	@Test
+	public void testNotRetrieveHostName() {
+		try {
+			InetAddress address = mock(InetAddress.class);
+			when(address.getCanonicalHostName()).thenReturn("worker10");
+			when(address.getHostName()).thenReturn("worker10");
+			when(address.getHostAddress()).thenReturn("127.0.0.1");
+
+			TaskManagerLocation info = new TaskManagerLocation(ResourceID.generate(), address, 19871, false);
+			assertNotEquals("worker10", info.getHostname());
+			assertNotEquals("worker10", info.getFQDNHostname());
+			assertEquals("127.0.0.1", info.getHostname());
+			assertEquals("127.0.0.1", info.getFQDNHostname());
+		}
+		catch (Exception e) {
+			e.printStackTrace();
+			fail(e.getMessage());
```

Comment:
You don't need to manually fail the test. if there's an exception thrown out of the test, it'll fail.
The other code in this file is very old, and uses an outdated pattern 😄